api-reference/10 UI Components/Errors and Warnings/W1018.md

```diff
@@ -0,0 +1,20 @@
+---
+id: ErrorsUIWidgets.W1018
+---
+---
+##### shortDescription
+Occurs in the DataGrid when infinite scrolling is used and **selection**.[allowSelectAll](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/selection/#allowSelectAll) is set to **true**.
+
+---
+To select all rows, the component requires their total number. With infinite scrolling, the component cannot calculate it because rows are loaded dynamically.
```

Comment:
1) `To select all rows, the component requires their total number.`

What do you mean by "their"? Please specify.

2) `With infinite scrolling, the component cannot calculate it because rows are loaded dynamically.`

We should avoid the gerund, and specify _who_ scrolls. Is it the user?

Also, what do you mean by "calculate it"? What are you calculating?

Reply:
"their total number" means the total number of rows.
"the component cannot calculate" means that the component can't count rows to get the total number of rows.

> We should avoid the gerund, and specify who scrolls. Is it the user?
Replied in the comment for line #6.

Reply:
```suggestion
To select all rows, the component requires their total number. The component cannot calculate the total number when infinite scrolling is used because rows are loaded dynamically.
```